perl/ANNOTATION.md

```diff
@@ -0,0 +1,781 @@
+
+3. **Insertions and Deletions**: For transcript-level annotations like `refSeq.*` (refSeq transcript annotations), `nearest.*`, and `nearestTss.*`, insertions and deletions affecting multiple nucleotides are separated by pipes (`|`). This allows reporting a transcript consequence per disrupted base.
+
+4. **Reserved Delimiters**: Reserved delimiters like `;`, `/`, and `|` are essential for maintaining the structure of relationships in the data. If these characters appear in the source data, they are replaced with commas to ensure they can still be used effectively as delimiters in the output.
+
+This structure enables precise retention of complex data relationships across multiple annotation fields while maintaining a highly structured and parseable output.
+
+#### What Information Can Bystro Output?
+
+Bystro is a general-purpose annotation engine, and has no restrictions on the types of annotations/feature labels it can output. The annotations are sourced from the Bystro Database chosen, and is described based on a YAML configuration file, which describes the fields to be output, and the sources of the annotations.
+
+A Bystro Database is a high-performance key-value database that uses the Lightning Memory Map Database (LMDB) engine. It supports millions of lookups per second on a single machine, and can be used to store and retrieve annotations for millions of variants.
+
+Bystro Databases can be re-created from the YAML configuration file corresponding to that database, and new databases with different information can be created by editing the YAML configuration file, and re-running the Bystro Database creation process.
+
+#### Variant Representation
+
+Bystro's variant representation deviates slightly from the standard VCF format. Namely, Bystro ensures that the reference base is always a single nuleotide, and does the appropriate normalization/transformation of input variants to ensure this. This variant representation greatly simplifies downstream analysis.
+
+Compared with the VCF format, Bystro's variant representation has the following differences:
+
+SNPs: Unchanged
+
+Insertions: The reference base is always the base just before the insertion, and the alternate allele is the inserted sequence, preceeded by a +. The position is the first affected base.
+
+- For instance, if the VCF had POS: 1 REF: AT and ALT: ACCT, Bystro will represent this as inputRef: T, alt: +CCT, pos: 1, with the position the same as that in the VCF file, which is the first position of the reference base (the base just before the first inserted base). In Bystro's case there is only 1 reference base always, so the position is that of the only affected base.
```

Comment:
For visual clarity, I think I would break the examples out into code block lines throughout this explanation, and same for the one below.